packages/editor/src/plugins/image/components/pixabay-image-search.tsx

```diff
@@ -0,0 +1,158 @@
+  const imageStrings = editorStrings.plugins.image
+  const apiKey = 'secret'
+
+  useEffect(() => {
+    if (isLoadingImage) {
+      document.body.classList.add('overflow-hidden')
+    } else {
+      document.body.classList.remove('overflow-hidden')
+    }
+  }, [isLoadingImage])
+
+  const handleSearch = async (searchQuery: string) => {
+    if (!searchQuery) return
+    setIsSearching(true)
+    try {
+      const response = await fetch(
+        `https://pixabay.com/api/?key=${apiKey}&q=${encodeURIComponent(searchQuery)}&image_type=photo`
+      )
+      const data = (await response.json()) as PixabayResponse
+      setImages(data.hits)
+      setShowTags(false)
+    } catch (error) {
+      setShowTags(true)
+      console.error('Error fetching images from Pixabay:', error)
+    } finally {
+      setIsSearching(false)
+    }
+  }
+
+  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setQuery(e.target.value)
+  }
+
+  const handleTagClick = async (tag: string) => {
+    setQuery(tag)
+    await handleSearch(tag)
+  }
+
+  const handleKeyDown = async (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key !== 'Enter') return
+    await handleSearch(query)
+  }
+
+  return (
+    <div className="min-h-[60vw] pt-10">
+      <h2 className="mb-6 ml-10 mt-10 font-bold">{imageStrings.licenceFree}</h2>
+      {/* Search input */}
+      <div className="w-full">
+        <input
+          type="text"
+          value={query}
+          onChange={handleInputChange}
+          placeholder="Suche"
+          onKeyDown={handleKeyDown}
+          className="ml-10 w-[90%] rounded rounded-lg border border-0 bg-yellow-100 p-2 px-4 py-2 text-gray-600"
+        />
+      </div>
+
+      {/* Search tags */}
+      {showTags && (
+        <div>
+          <div className="mb-6 mt-10 flex flex-wrap  justify-center">
+            {Object.values(imageStrings.searchTags).map((tagKey) => (
+              <button
+                key={tagKey}
+                onClick={() => handleTagClick(tagKey)}
+                className="m-2 inline-block rounded-md bg-white px-4 py-2 text-black shadow-md"
+              >
+                {tagKey}
+              </button>
+            ))}
+          </div>
+        </div>
+      )}
+      <div
+        className={cn(
+          'max-h-[500px]',
+          'mt-4 flex flex-wrap px-8 ',
+          isLoadingImage && 'max-h-100',
+          isLoadingImage && 'border-1 border border-red-500'
+        )}
+        style={{ overflow: isLoadingImage ? 'hidden' : 'auto' }}
+      >
+        {/* Loading overlay */}
+        {(isLoadingImage || isSearching) && (
+          <div className="absolute inset-0 flex items-center justify-center bg-white bg-opacity-75 backdrop-blur-sm">
+            <h1 className="text-center text-xl font-bold">
+              {isLoadingImage
+                ? imageStrings.loadingImage
+                : isSearching
+                  ? imageStrings.searching
+                  : 'Hmm'}
```

Comment:
If "Hmm" is never reached, you could also remove the last conditional, right?

```suggestion
              {isLoadingImage
                ? imageStrings.loadingImage
                : imageStrings.searching }
```

Reply:
Great, thanks, done